blog/2024-05-01-alternatives-auth0.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "Keycloak vs. Auth0, a Open-Source Alternative"
+slug: keycloak-vs-auth0-open-source-alternative
+date: 2020-05-01
+author: Phase Two
+tags: [phase_two, open_source, authentication, authorization, auth0, keycloak]
+draft: true
+---
+
+### Exploring Keycloak as an Alternative to Auth0 for Authentication Solutions
+
+When it comes to implementing authentication and authorization in web applications, Auth0 and Keycloak are two prominent solutions that offer robust security features. While Auth0 is a popular choice for many developers due to its comprehensive, cloud-based platform, Keycloak presents a compelling alternative, especially in terms of cost and flexibility. This blog post will delve into how Keycloak stacks up against Auth0, focusing on cost of ownership, maintenance, and functionality.
+
+#### Cost of Ownership
+
+There are many factors to think about with a total cost-of-ownership. We've written extensively about [choosing an Open-Source IAM](./2024-03-11-open-source-iam.md) over a managed service.
+
+**Auth0:**
+Auth0 operates on a subscription model, which can be quite cost-effective for startups or small projects with minimal authentication needs. It offers a free tier limited by the number of authentications per month and features. Beyond this, prices increase based on the number of users and the level of features required. For large enterprises, this can translate into a significant ongoing expense.
```

Comment:
I think we should be much more clear about this. Auth0 users frequently get sticker shock after they exceed the free tier, and are faced with 100k+ bills for a modest number of users and idp connections.